File: Java/CodingBat/xyzThere.java
// http://codingbat.com/prob/p136594

public boolean xyzThere(String str) {
  int xyzIndex = str.indexOf("xyz");
  
  while (xyzIndex != -1) {
    if (xyzIndex == 0) {
      return true;
    }
    else if (str.charAt(xyzIndex-1) != '.') {
      return true;
    }
    xyzIndex = str.indexOf("xyz", xyzIndex+2);
  }
  
  return false;
}

File: Python/CodingBat/xyz_there.py
# http://codingbat.com/prob/p149391

def xyz_there(str):
  for i in range(len(str)-2):
    if (str[i:i+3] == "xyz"):
      if ( i == 0 or (i != 0 and str[i-1] != '.') ):
        return True
  
  return False
